### ISSUE TITLE
Unexpected `FunctionClauseError` when raising custom error

### ISSUE THREAD

**Issue by axelson** [jvoegele/errata#1]
Hi! 👋
Thank you for this library! I'm excited to use it because I like the idea of structured errors. However, I'm encountering an error when I try to use this with `raise`.

Given a custom error defined like:
```elixir
defmodule ErrataRepro.Error do
  use Errata.Error
end
```

I'd expect to be able to raise it like:
```elixir
path = "my_path.txt"
script = "some_script"

raise ErrataRepro.Error, reason: :script_not_found, context: %{path: path, script: script}
```

However that results in a `FunctionClauseError`:
```
** (ErrataRepro.Error) got FunctionClauseError with message "no function clause matching in Errata.Errors.format_message/1" while retrieving Exception.message/1 for %ErrataRepro.Error{__errata_error__: true, kind: :general, message: nil, reason: :script_not_found, context: %{path: "my_path.txt", script: "some_script"}, env: nil}. Stacktrace:
    (errata 0.8.0) lib/errata/errors.ex:36: Errata.Errors.format_message(%ErrataRepro.Error{__errata_error__: true, kind: :general, message: nil, reason: :script_not_found, context: %{path: "my_path.txt", script: "some_script"}, env: nil})
    (elixir 1.17.2) lib/exception.ex:85: Exception.message/1
    (elixir 1.17.2) lib/exception.ex:136: Exception.format_banner/3
    (iex 1.17.2) lib/iex/evaluator.ex:404: IEx.Evaluator.print_error/3
    (iex 1.17.2) lib/iex/evaluator.ex:298: IEx.Evaluator.parse_eval_inspect/4
    (iex 1.17.2) lib/iex/evaluator.ex:187: IEx.Evaluator.loop/1
    (iex 1.17.2) lib/iex/evaluator.ex:32: IEx.Evaluator.init/5
    (stdlib 5.2.3.1) proc_lib.erl:241: :proc_lib.init_p_do_apply/3

    (errata_repro 0.1.0) lib/errata_repro.ex:6: ErrataRepro.raise_error/0
    iex:1: (file)
```

Here's a reproduction: https://github.com/axelson/errata_repro

Calling it like:
```elixir
path = "my_path.txt"
script = "some_script"

raise ErrataRepro.Error, reason: :script_not_found, context: %{path: path, script: script}, message: "some message"
```
Results in:
```
** (ErrataRepro.Error) some message: :script_not_found
    (errata_repro 0.1.0) lib/errata_repro.ex:6: ErrataRepro.raise_error/0
    iex:2: (file)
```
Which is more reasonable, but it is maybe weird that my message wasn't used at all.

**Comment by jvoegele** [comment]
Thank you @axelson, I will look into this at first opportunity.

**Comment by jvoegele** [comment]
> Calling it like:
> 
> ```elixir
> path = "my_path.txt"
> script = "some_script"
> 
> raise ErrataRepro.Error, reason: :script_not_found, context: %{path: path, script: script}, message: "some message"
> ```
> 
> Results in:
> 
> ```
> ** (ErrataRepro.Error) some message: :script_not_found
>     (errata_repro 0.1.0) lib/errata_repro.ex:6: ErrataRepro.raise_error/0
>     iex:2: (file)
> ```
> 
> Which is more reasonable, but it is maybe weird that my message wasn't used at all.

Just to make a note about this point, the message that you passed to `raise` in the above example ("some message") actually is displayed in the exception report:


> ** (ErrataRepro.Error) **some message**: :script_not_found

**Comment by axelson** [comment]
> Just to make a note about this point, the message that you passed to raise in the above example ("some message") actually is displayed in the exception report

Ah, whoops! I totally missed it in there. I probably should've used a less generic message for testing 😅 

Thank you for taking a look!

**Comment by jvoegele** [comment]
Fixed in version `0.8.1`.

Here is the commit for reference: https://github.com/jvoegele/errata/commit/fd66ccf514d8ba17b288ee3dd3032f60cd105414

**Comment by axelson** [comment]
Thank you for the fix! I'm excited about this library 🎉
